Orange/widgets/evaluate/owparameterfitter.py

```diff
@@ -318,93 +368,150 @@ def _add_plot(self):
         box.layout().addWidget(self.graph)
 
     def _add_controls(self):
-        box = gui.vBox(self.controlArea, "Settings")
-        self.__combo = gui.comboBox(box, self, "parameter_index",
+        layout = QGridLayout()
+        gui.widgetBox(self.controlArea, "Settings", orientation=layout)
+        self.__combo = gui.comboBox(None, self, "parameter_index",
                                     model=self.__parameters_model,
                                     callback=self.__on_parameter_changed)
-
-        buttons = gui.radioButtons(box, self, "type",
-                                   callback=self.__on_setting_changed)
-
-        gui.appendRadioButton(buttons, "Range")
-        hbox = gui.indentedBox(buttons, 20, orientation=Qt.Horizontal)
-        kw = {"minv": -MIN_MAX_SPIN, "maxv": MIN_MAX_SPIN,
-              "callback": self.__on_setting_changed}
-        self.__spin_min = gui.spin(hbox, self, "minimum", label="Min:", **kw)
-        self.__spin_max = gui.spin(hbox, self, "maximum", label="Max:", **kw)
-
-        gui.appendRadioButton(buttons, "Manual")
-        hbox = gui.indentedBox(box, 20, orientation=Qt.Horizontal)
-        gui.lineEdit(hbox, self, "manual_steps", placeholderText="10, 20, 30",
-                     callback=self.__on_setting_changed)
-
-        box = gui.vBox(self.controlArea, "Steps preview")
-        self.preview = ""
-        gui.label(box, self, "%(preview)s", wordWrap=True)
+        layout.addWidget(self.__combo, 0, 0, 1, 2)
+
+        buttons = gui.radioButtons(None, self, "type",
+                                   callback=self.__on_type_changed)
+        button = gui.appendRadioButton(buttons, "Range:")
+        layout.addWidget(button, 1, 0)
+
+        kw = dict(minv=-MIN_MAX_SPIN, maxv=MIN_MAX_SPIN,
+                  alignment=Qt.AlignRight,
+                  callback=self.__on_min_max_changed)
+        box = gui.hBox(None)
+        self.__spin_min = gui.spin(box, self, "minimum", label="From:", **kw)
+        layout.addWidget(box, 1, 1)
+
+        box = gui.hBox(None)
+        self.__spin_max = gui.spin(box, self, "maximum", label="To:", **kw)
+        layout.addWidget(box, 2, 1)
+
+        self.range_preview = RangePreview()
+        layout.addWidget(self.range_preview, 3, 0, 1, 2)
+
+        gui.appendRadioButton(buttons, "Manual:")
+        layout.addWidget(buttons, 4, 0)
+        edit = gui.lineEdit(None, self, "manual_steps",
+                            placeholderText="10, 20, 30",
+                            callback=self.__on_manual_changed)
+        layout.addWidget(edit, 4, 1)
+
+        # gui.lineEdit's connect does not call the callback on return pressed
+        # if the line hasn't changed.
+        @edit.returnPressed.connect
+        def _():
+            self.type = self.MANUAL
```

Comment:
The thing with this line edit is: if the user changes it and presses return, we want a recomputation. The `callback` takes care of this.

Now imagine that the user enters some data into the line edit. Then he clicks the first radion (`FROM_RANGE`). Now he clicks on the line edit. Nothing happens (as it shouldn't). Then he presses return (without changing anything). The callback is not triggered because nothing is change. But we still want to change the type to manual. Hence this additional connect.

We need to check, though, that recomputation it is not triggered twice.